### DISCUSSION TITLE
Does ActivityWatch do keyboard key-strokes statistics?

### DISCUSSION THREAD

**Discussion by jakimfett** [ActivityWatch/activitywatch#619] [Q&A] [answered]
Being the preferred sleeping-spot of our 3-month-old has me considering the use of an one-handed (combo/chorded-capable) coding keyboard, and as a friend has offered me some test hardware (thanks @kemonine) it's time for me to figure out which of my keys are most used during my increasingly-short times in full use of both hands while behind the keyboard at my desk.

To this end, my queries:
 - Does ActivityWatch give the user visibility (or exportability) on the keystroke data?

(Having a keyboard-heatmap, and display of the number of times the individual key was used, would be the ideal.)

It's relevant that ![WhatPulse](https://qph.fs.quoracdn.net/main-qimg-dee0df9a02e38234c17e97bb79f272c9) is almost close enough for me to risk something centralized watching my literal every keystroke as a system administrator.

(The cognitive load and broken work/thoughtflow of needing to disable before and re-enable after every time I'm throwing protected information at the screen is...not palatable.)

So.

ActivityWatch.

Does it have what a decentralized-oriented, privacy-conscious sysadmin such as myself needs?

**Comment by ErikBjare** [comment, marked as answer]
The closest thing we have is [aw-watcher-input](https://github.com/ActivityWatch/aw-watcher-input), but it does not record *which* key is being pressed (due to the security concern and risks of being used by bad actors as a keylogger), although it shouldn't be very difficult to modify it to suit your needs.
